client/app/account/user/agreements/user-agreements.service.js

```diff
@@ -25,11 +90,24 @@ angular.module("UserAccount").service("UserAccount.services.agreements", [
                     },
                     serviceType: "aapi"
                 })
+                .then(formatList)
                 .then(getSuccessDataOrReject);
         };
 
         this.getAgreement = function (agreementId) {
-            return $http.get(`${proxyPath}/agreements/${agreementId}`).then(getSuccessDataOrReject);
+            return $http.get(`${proxyPath}/agreements/${agreementId}`)
+                .then((response) => {
+                    if (response.data && response.data.contractId) {
```

Comment:
`_(response).has("data.contractId")`

Reply:
Even if this would be a little bit more clear, I'm trying to code as much as possible in pure ES6